app/src/main/contrib/components/Responsibilities/BurdenEstimates/UploadBurdenEstimatesForm.tsx

```diff
@@ -1,86 +1,345 @@
 import * as React from "react";
+import {FormEvent} from "react";
 import {Alert} from "reactstrap";
-import {BurdenEstimateSet, ErrorInfo} from "../../../../shared/models/Generated";
-import {CreateBurdenEstimateSetForm} from "./CreateBurdenEstimateSetForm";
-import {PopulateEstimatesForm} from "./PopulateBurdenEstimatesForm";
-import {Dispatch} from "redux";
+import {checkFileExtensionIsCSV, CustomValidationResult} from "../../../../shared/validation/FileValidationHelpers";
+import {
+    BurdenEstimateSetType,
+    BurdenEstimateSetTypeCode,
+    ErrorInfo,
+    Result
+} from "../../../../shared/models/Generated";
+import {ConfigurationHash, Resumable, ResumableFile} from "./ResumableTypes";
+import {settings} from "../../../../shared/Settings";
 import {ContribAppState} from "../../../reducers/contribAppReducers";
+import {Dispatch} from "redux";
 import {estimatesActionCreators} from "../../../actions/estimatesActionCreators";
 import {connect} from "react-redux";
+import {branch, compose, renderComponent} from "recompose";
+import {LoadingElement} from "../../../../shared/partials/LoadingElement/LoadingElement";
+import {roundToOneDecimalPlace} from "../../../../shared/Helpers";
+import {OptionSelector} from "../../OptionSelector/OptionSelector";
+
+const FileUploadClient = require('resumablejs');
 
-export interface UploadBurdenEstimatesFormPublicProps {
+export interface UploadEstimatesState {
+    file: ResumableFile,
+    progress: number,
+    isUploading: boolean,
+    uploadErrors: ErrorInfo[],
+    fileValidationResult: CustomValidationResult,
+    metadata: BurdenEstimateSetType
+}
+
+export interface UploadEstimatesPublicProps {
     touchstoneId: string;
     scenarioId: string;
     groupId: string;
-    estimateSet: BurdenEstimateSet | null;
-    canUpload: boolean;
-    canCreate: boolean;
 }
 
-export interface UploadBurdenEstimatesFormComponentProps extends UploadBurdenEstimatesFormPublicProps {
-    hasSuccess: boolean;
-    errors: ErrorInfo[];
+export interface UploadEstimatesProps extends UploadEstimatesPublicProps {
+    createUploadClient: (settings: ConfigurationHash) => Resumable,
+    populateEstimateSet: (token: string) => void,
     resetPopulateState: () => void;
+    url: string;
+    uploadToken: string;
+    createBurdenEstimateSet: (data: BurdenEstimateSetType) => void;
```

Comment:
The change here is that `requestUploadToken` has become `createBurdenEstimateSet` (which itself creates the set and then gets the upload token)